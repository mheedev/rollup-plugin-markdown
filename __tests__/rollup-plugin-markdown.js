const rollup = require('rollup');
const path = require('path');
const Module = require('module');
const markdownPlugin = require('../src/index');

function moduleFromString(code) {
  const opts = {};
  const filename = 'module_from_string.js';

  opts.appendPaths = opts.appendPaths || [];
  opts.prependPaths = opts.prependPaths || [];

  if (typeof code !== 'string') {
    throw new Error('code must be a string');
  }

  let paths = Module._nodeModulePaths(path.dirname(filename));

  let parent = module.parent;
  let m = new Module(filename, parent);
  m.filename = filename;
  m._compile(code, filename);

  let exports = m.exports;
  parent && parent.children && parent.children.splice(parent.children.indexOf(m), 1);

  return exports;
}

process.chdir(__dirname);

const bundleFile = async rollupConfig => {
  const bundle = await rollup.rollup(rollupConfig);

  const { output } = await bundle.generate({ format: 'cjs' });

  const [{ code }] = output;
  return code;
};

it('returns a module for the markdown file', async () => {
  const code = await bundleFile({
    input: '../fixtures/test.md',
    plugins: [markdownPlugin()],
  });

  const requiredModule = moduleFromString(code);

  expect(requiredModule.html).toMatchSnapshot();
  expect(requiredModule.filename).toEqual('test.md');
  expect(requiredModule.path).toEqual(path.resolve(path.join(__dirname, 'fixtures/test.md')));
});