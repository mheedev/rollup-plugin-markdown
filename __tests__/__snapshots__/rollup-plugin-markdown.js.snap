// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns a module for the markdown file 1`] = `
"<ul>
<li>This is a title</li>
</ul>
<pre class=\\"shiki\\" style=\\"background-color: #2e3440\\"><code><span style=\\"color: #81A1C1\\">const</span><span style=\\"color: #D8DEE9FF\\"> </span><span style=\\"color: #D8DEE9\\">text</span><span style=\\"color: #D8DEE9FF\\"> </span><span style=\\"color: #81A1C1\\">=</span><span style=\\"color: #D8DEE9FF\\"> </span><span style=\\"color: #ECEFF4\\">\`</span><span style=\\"color: #A3BE8C\\">This is a code block</span><span style=\\"color: #ECEFF4\\">\`</span><span style=\\"color: #81A1C1\\">;</span>
<span style=\\"color: #8FBCBB\\">console</span><span style=\\"color: #ECEFF4\\">.</span><span style=\\"color: #88C0D0\\">log</span><span style=\\"color: #D8DEE9FF\\">(</span><span style=\\"color: #D8DEE9\\">text</span><span style=\\"color: #D8DEE9FF\\">)</span><span style=\\"color: #81A1C1\\">;</span></code></pre>
<p>This is another paragraph</p>
<p>And <a href=\\"www.google.com\\">here</a> we have a link</p>
"
`;
